src/app/auth/profiles/profiles.component.ts

```diff
@@ -92,14 +61,11 @@ export class ProfilesComponent implements OnInit {
     fileDirectory: null
   };
 
-  constructor(public errorDialog: MatDialog) { }
+  constructor(private electronService: ElectronService, public errorDialog: MatDialog) { }
 
   ngOnInit() {
-    const path = require('path');
-    const temp = ipcRenderer.sendSync('synchronous-message', 'getDirectory');
-    this.filePath = path.join(
-        temp.replace(this.APPLICATION_AND_SUBDIRECTORIES, ''),
-        this.PROFILES_FILE_NAME);
+    const temp = this.electronService.sendIpcSycMessage('synchronous-message', 'getDirectory');
```

Comment:
Minor refactoring suggestion: Instead of using `sendIpcSycMessage` here, can the electron service expose a method for retrieving the app directory? (Use `sendIpcSycMessage` within this method).